guides/source/configuring.md

```diff
@@ -2762,6 +2763,46 @@ class EncoderJob < ActiveJob::Base
 end
 ```
 
+#### `config.active_job.enqueue_after_transaction_commit`
+
+Controls whether Active Job #perform_later and similar methods automatically defer
+the job queuing to after the current Active Record transaction is committed.
+
+It can be set to `:never` to never defer the enqueue, to `:always` always defer
+the enqueue, or to `:auto` to let the queue adapter decide if it should be defered
+or not. Active Job backends that use the same database than Active Record as a queue,
+should generally prevent the deferring, and others should allow it.
+
+Example:
+
+```ruby
+Topic.transaction do
+  topic = Topic.create(title: "New Topic")
+  NewTopicNotificationJob.perform_later(topic)
+end
+```
+
+In this example, if enqueue after transaction commit is enabled, the job will
+be enqueued immediately, even thought the `Topic` hasn't been committed yet.
+Because of this, if the job is picked up almost emmediately, it will fail to
```

Comment:
Or if the transaction rolls back after the job is enqueued (e.g. we could lose the connection when trying to COMMIT)